lib/CreateChangesetUpdateChangelog.js

```diff
@@ -5,14 +5,25 @@ const replace = require('replace-in-file');
 module.exports = class {
     constructor() {
     }
-    createChangeset(dbChange,changesetPath,rfcNo,version){        
-        var tDate=moment.tz(Date.now(),'America/Vancouver').format('YYYYMMDDHHmmss')
-        var newFileName=changesetPath+'/'+'V'+tDate+version+'__'+rfcNo+'_'+dbChange+'.sql'
-        fs.open(newFileName, 'w', function(err,file){
-            if(err) throw err;
-                console.log('File '+newFileName+' was created successfully')
-            }); 
-        return newFileName
+   createChangeset(typeOfMigration,dbChange,changesetPath,version){ 
+        if(typeOfMigration.toLowerCase()=='versioned'){       
+            var tDate=moment.tz(Date.now(),'America/Vancouver').format('YYYYMMDDHHmmss')
+            var newFileName=changesetPath+'/'+'V'+tDate+version+'__'+'_'+dbChange +'.sql'
+            fs.open(newFileName, 'w', function(err,file){
+                if(err) throw err;
+                }); 
+            return newFileName
+        }
+        else if(typeOfMigration.toLowerCase()=='repeatable'){
+            var newFileName=changesetPath+'/'+'R'+'__'+dbChange+'.sql'
+            fs.open(newFileName, 'w', function(err,file){
+                if(err) throw err;
```

Comment:
I am not too sure, but you might need to use fs.close to close the 'file' when using fs.open.
Could you check document?
Thanks.

Reply:
I dont think it is a hard and fast rule since we need to close it only if we need to use the buffer or it gets auto closed. But i agree it is good to practise to do it